File: ios/GodotViewManager.m
#import <React/RCTViewManager.h>
#import <React/RCTUIManager.h>

@interface RCT_EXTERN_MODULE(GodotViewManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(projectPath, NSString)
RCT_EXPORT_VIEW_PROPERTY(mainScene, NSString)
RCT_EXPORT_VIEW_PROPERTY(symbolPrefix, NSString)
RCT_EXPORT_VIEW_PROPERTY(symbolPrefixes, NSString) // NOWY
RCT_EXPORT_VIEW_PROPERTY(suppressStubLogs, NSNumber)
RCT_EXPORT_VIEW_PROPERTY(autoStart, NSNumber)
RCT_EXTERN_METHOD(sendEventToGodot:(nonnull NSNumber *)reactTag event:(NSString *))
RCT_EXTERN_METHOD(ensureEngine:(nonnull NSNumber *)reactTag)
RCT_EXTERN_METHOD(diagnoseStub:(nonnull NSNumber *)reactTag)
RCT_EXTERN_METHOD(setScene:(nonnull NSNumber *)reactTag scene:(NSString *))   // dodane
RCT_EXTERN_METHOD(forceAttachRenderView:(nonnull NSNumber *)reactTag) // NOWE
@end

File: ios/GodotViewManager.swift
import Foundation
import React

@objc(GodotViewManager)
class GodotViewManager: RCTViewManager {

  override static func requiresMainQueueSetup() -> Bool {
      return true
  }

  override func view() -> NSView! {
      return GodotView(frame: .zero)
  }

  @objc func sendEventToGodot(_ reactTag: NSNumber, event: NSString) {
      DispatchQueue.main.async {
          if let view = self.bridge.uiManager.view(forReactTag: reactTag) as? GodotView {
              view.receiveEventFromReact(event: event as String)
          }
      }
  }
}
